app/gateway-oss/2.4.x/plugin-development/custom-logic.md

```diff
@@ -26,47 +26,47 @@ kong.plugins.<plugin_name>.handler
-| `:body_filter()`   | [body_filter]     | Executed for each chunk of the response body received from the upstream service. Since the response is streamed back to the client, it can exceed the buffer size and be streamed chunk by chunk. hence this method can be called multiple times if the response is large. See the [lua-nginx-module] documentation for more details.
-| `:log()`           | [log]             | Executed when the last response byte has been sent to the client.
+| Function name   | Phase             | Description
+|-----------------|-------------------|------------
+| `init_worker`   | [init_worker]     | Executed upon every Nginx worker process's startup.
+| `certificate`   | [ssl_certificate] | Executed during the SSL certificate serving-phase of the SSL handshake.
+| `rewrite`       | [rewrite]         | Executed for every request upon its reception from a client as a rewrite phase handler. *NOTE:* In this phase, neither the `Service` nor the `Consumer` have been identified, hence this handler is only executed if the Plugin was configured as a global Plugin.
+| `access`        | [access]          | Executed for every request from a client before it is being proxied to the upstream service.
+| `response` | [access] | Replaces both `header_filter()` and `body_filter()`. Executed after the whole response has been received from the upstream service, but before sending any part of it to the client.
+| `header_filter` | [header_filter]   | Executed when all response headers bytes have been received from the upstream service.
+| `body_filter`   | [body_filter]     | Executed for each chunk of the response body received from the upstream service. Because the response is streamed back to the client, it can exceed the buffer size and be streamed chunk by chunk. This method can be called multiple times if the response is large. See the [lua-nginx-module] documentation for more details.
+| `log`           | [log]             | Executed when the last response byte has been sent to the client.
 
 **Note:**
 
-If a module implements the `:response()` method, Kong will automatically activate the "buffered proxy" mode, as if the [`kong.service.request.enable_buffering()` function][enable_buffering] had been called.  Because of a current Nginx limitation, this doesn't work for HTTP/2 or gRPC upstreams.
+If a module implements the `:response()` method, {{site.ce_product_name}} automatically activates the buffered proxy mode, as if the [`kong.service.request.enable_buffering()` function][enable_buffering] had been called.  Because of a current Nginx limitation, this doesn't work for HTTP/2 or gRPC upstreams.
 
-To reduce unexpected behaviour changes, Kong will abort startup if a plugin implements both `:response()` and either `:header_filter()` or `:body_filter()`.
+To reduce unexpected behaviour changes, {{site.ce_product_name}} aborts startup if a Plugin implements both `:response()` and either `:header_filter()` or `:body_filter()`.
 
```

Comment:
shouldn't these notes also be changed to 'function' style, instead of 'method'?

Reply:
I have changed everything to `function`, and called everything a "function" instead of a "method"